assimpTargets.cmake computes bad .dylib filename on macOS

I have assimp 5.0.0 installed on macOS Mojave via homebrew. There seems to be something wrong with the installed CMake configuration files, leading to generation of an almost-but-not-quite-right path for the assimp shared lib.

The installed CMake rules want to find `/usr/local/lib/libassimp.dylib.5`, while the installed filename is `/usr/local/lib/libassimp.5.dylib`.

Workaround: `cd /usr/local/lib; ln -s libassimp.5.dylib libassimp.dylib.5`.

Repro:

1. Put this content in a file called `CMakeLists.txt`:
```cmake_minimum_required(VERSION 3.5)
project(foo)
find_package(assimp)
```
2. Run `cmake .`
3. Get this error:
```CMake Error at /usr/local/lib/cmake/assimp-5.0/assimpTargets.cmake:85 (message):
  The imported target "assimp::assimp" references the file

     "/usr/local/lib/libassimp.dylib.5"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/local/lib/cmake/assimp-5.0/assimpTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/local/lib/cmake/assimp-5.0/assimp-config.cmake:1 (include)
  CMakeLists.txt:4 (find_package)
```

--- a/assimpTargets-release.cmake.in
+++ b/assimpTargets-release.cmake.in
@@ -63,7 +63,11 @@ if(MSVC)
 else()
   set(ASSIMP_LIBRARY_SUFFIX "@ASSIMP_LIBRARY_SUFFIX@" CACHE STRING "the suffix for the assimp libraries" )
   if(ASSIMP_BUILD_SHARED_LIBS)
-    set(sharedLibraryName "libassimp${ASSIMP_LIBRARY_SUFFIX}@CMAKE_SHARED_LIBRARY_SUFFIX@.@ASSIMP_VERSION_MAJOR@")
+    if(APPLE)
+      set(sharedLibraryName "libassimp${ASSIMP_LIBRARY_SUFFIX}.@ASSIMP_VERSION_MAJOR@@CMAKE_SHARED_LIBRARY_SUFFIX@")
+    else()
+      set(sharedLibraryName "libassimp${ASSIMP_LIBRARY_SUFFIX}@CMAKE_SHARED_LIBRARY_SUFFIX@.@ASSIMP_VERSION_MAJOR@")
+    endif()
     set_target_properties(assimp::assimp PROPERTIES
       IMPORTED_SONAME_RELEASE "${sharedLibraryName}"
       IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/${sharedLibraryName}"

